api/v1beta1/kafkacluster_types.go

```diff
@@ -226,39 +251,100 @@ type BrokerConfig struct {
 	// +kubebuilder:validation:Enum=Hostname;ExternalIP;InternalIP;InternalDNS;ExternalDNS
 	// +optional
 	NodePortNodeAddressType corev1.NodeAddressType `json:"nodePortNodeAddressType,omitempty"`
-	// Any definition received through this field will override the default behaviour of OneBrokerPerNode flag
-	// and the operator supposes that the user is aware of how scheduling is done by kubernetes
-	// Affinity could be set through brokerConfigGroups definitions and can be set for individual brokers as well
-	// where letter setting will override the group setting
-	Affinity           *corev1.Affinity           `json:"affinity,omitempty"`
-	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
-	// SecurityContext allows to set security context for the kafka container
-	SecurityContext *corev1.SecurityContext `json:"securityContext,omitempty"`
-	// BrokerIngressMapping allows to set specific ingress to a specific broker mappings.
-	// If left empty, all broker will inherit the default one specified under external listeners config
-	// Only used when ExternalListeners.Config is populated
-	BrokerIngressMapping []string `json:"brokerIngressMapping,omitempty"`
-	// InitContainers add extra initContainers to the Kafka broker pod
-	InitContainers []corev1.Container `json:"initContainers,omitempty"`
-	// Containers add extra Containers to the Kafka broker pod
+}
+
+// Controller represents "controller" nodes in KRaft. This is not applicable to ZooKeeper mode
+type Controller struct {
```

Comment:
The final version of `Controller` subjects to change if we decide to deploy the controllers as `StatefulSet`